### PR TITLE
Add avatar uploading

### LINKED ISSUE #8: Allow users to upload Avatars
This one's fairly self explanatory. We'll be using DigitalOcean Spaces for this, via the [Offprint CDN](https://images.offprint.net/). 

NestJS also has some good documentation on multipart parsing, [available here](https://docs.nestjs.com/techniques/file-upload).

### PR DESCRIPTION
**Pull request by pingzing**
Closes #8.

boof

And recorded here for posterity: Mongoose's `findOneAndUpdate()` returns the _original_  document, not the updated one. This probably causes issues in some other places too.

### CODE REVIEW

File: frontend/src/app/models/site/httpError.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export interface HttpError {
```
Comment (pingzing) [inline frontend/src/app/models/site/httpError.ts:1]: (please tell me this doesn't already exist somewhere)

File: frontend/src/app/pages/home/settings/settings.component.less [hunk 2]
```diff
@@ -47,12 +47,13 @@ div.settings-container {
...
                     img {
-                        max-width: 130px;
-                        max-height: 130px;
+                        width: 130px;
+                        height: 130px;
                         margin: 0 auto;
                         border-radius: 50%;
                         border: 1px solid var(--site-borders);
                         object-fit: cover;
+                        object-position: 50% 50%; // Determines the viewport position
```
Comment (pingzing) [inline frontend/src/app/pages/home/settings/settings.component.less:56]: ¯\\\_(ツ)_/¯

File: src/api/images/images.module.ts [hunk 3]
```diff
@@ -13,8 +12,7 @@ import { ImagesService } from './images.service';
         signOptions: { expiresIn: '3600s' },
       }),
     }),
-  ],
-  controllers: [ImagesController],
```
Comment (pingzing) [inline src/api/images/images.module.ts:17]: This was handy for debugging, but we don't need it.

**Review by Figments — APPROVED** [review]
Looks good!

**Pull request merged**